tests/erc20_base.rs

```diff
@@ -0,0 +1,152 @@
+    types::{Address, TransactionReceipt, U256},
+};
+use eyre::eyre;
+use std::io::{BufRead, BufReader};
+use std::str::FromStr;
+use std::sync::Arc;
+use tokio::sync::Mutex;
+use tokio::sync::OnceCell;
+
+/// deployer private key file path.
+const ALICE_PRIV_KEY_PATH: &str = "ALICE_PRIV_KEY_PATH";
+
+/// deployer private key file path.
+const BOB_PRIV_KEY_PATH: &str = "BOB_PRIV_KEY_PATH";
+
+/// Stylus RPC endpoint url.
+const RPC_URL: &str = "RPC_URL";
+
+/// Deployed program address.
+const MY_TOKEN_PROGRAM_ADDRESS: &str = "STYLUS_PROGRAM_ADDRESS";
+
+abigen!(
+    MyToken,
+    r#"[
+        function name() external view returns (string)
+        function symbol() external view returns (string)
+        function decimals() external view returns (uint8)
+        function totalSupply() external view returns (uint256)
+        function balanceOf(address account) external view returns (uint256)
+        function transfer(address recipient, uint256 amount) external returns (bool)
+        function allowance(address owner, address spender) external view returns (uint256)
+        function approve(address spender, uint256 amount) external returns (bool)
+        function transferFrom(address sender, address recipient, uint256 amount) external returns (bool)
+        function mint(address to, uint256 amount) external
+    ]"#
+);
+
+type MyTokenType = MyToken<SignerMiddleware<Provider<Http>, LocalWallet>>;
+type SignerClient = Arc<SignerMiddleware<Provider<Http>, LocalWallet>>;
+
+struct Fixtures {
+    alice_wallet: LocalWallet,
+    bob_wallet: LocalWallet,
+    alice_client: SignerClient,
+    bob_client: SignerClient,
+    erc20_token_address: Address,
+}
+
+static FIXTURES: OnceCell<Mutex<Fixtures>> = OnceCell::const_new();
+
+#[tokio::test]
+async fn mint_test() {
+    let fixtures_mutex = init_fixtures().await.unwrap();
+    let fixtures = fixtures_mutex.lock().await;
+
+    let alice_address = fixtures.alice_wallet.address();
+    let erc20_alice_signer =
+        MyToken::new(fixtures.erc20_token_address, fixtures.alice_client.clone());
+    let amount: U256 = 1000.into();
+
+    let alice_balance_before = balance_of(&erc20_alice_signer, alice_address).await.unwrap();
+    let _ = mint(&erc20_alice_signer, alice_address, amount).await.unwrap();
+    let alice_balance_after = balance_of(&erc20_alice_signer, alice_address).await.unwrap();
+
+    assert_eq!(alice_balance_after - alice_balance_before, amount);
+}
+
+/*** Erc20 helper functions ***/
+
+async fn balance_of(erc20_token: &MyTokenType, account: Address) -> eyre::Result<U256> {
+    let balance: U256 = erc20_token.balance_of(account).call().await?;
+    Ok(balance)
+}
+
+
+async fn mint(erc20_token: &MyTokenType, to: Address, amount: U256) -> eyre::Result<TransactionReceipt> {
+    let mint_tx: TransactionReceipt = erc20_token
+        .mint(to, amount)
+        .send()
+        .await
+        .unwrap()
+        .await
+        .unwrap()
+        .expect("Mint tx returned non");
```

Comment:
Typo?

Reply:
Also, why do we `expect` after an `unwrap`, is this a `Result<Result<...>>`?

Reply:
I fixed in last commit, converting Option directly to Result